### DISCUSSION TITLE
Minified JS file @octokit/rest ??

### DISCUSSION THREAD

**Discussion by npiccolo** [octokit/octokit.js#1745] [Q&A] [answered]
Hi all =D

Hope you are doing well !

I'm wondering where I can find a minified version for @octokit/rest. 
Or if you know a way to minified it =D

Thank you very much

Kind regards

**Comment by gr2m** [comment]
If you want to import `@octokit/rest` for usage in browsers, then you can do it like this

```html
<script type="module">
import { Octokit } from "https://cdn.pika.dev/@octokit/rest";
</script>
```

This will always import the latest version. It is already minified and optimized for usage in modern browsers.

If you want to avoid breaking changes in future, you can e.g. import only 17.x.x versions by using `https://cdn.pika.dev/@octokit/rest@17`

Does that work for you?

**Reply by npiccolo** [reply]
Thank you for the answer !!
This is not working for me, because I'm trying to integrate it into a Salesforce LWC Component.
I'm not able to import it like that, the system doesn't allow me to do this.
I think I need something like a 'octo-rest.min.js' file with all inside.

thank you very much for your help !

**Comment by gr2m** [comment]
I've attached a bundled & minified version of `@octokit/rest` to the latest release, including a `.map` file:
https://github.com/octokit/rest.js/releases/tag/v17.9.2

Does that work for you? I could set it up to do that automatically for all future releases, but you've been the first to ask for it so I'm not sure if it's worth the tradeoffs yet. 

I think there should also be online services that can bundle & minify npm packages on the fly, but I couldn't find any right now

**Reply by npiccolo** [reply]
I'll try that !!!
Thank you very much for the help really !!!
I know it's weird what I'm asking for, but those are the restrictions that I need to work with =P

BIG thank you !! =D

**Comment by npiccolo** [comment]
Hi again @gr2m ,

I was trying to integrate the minified file you have shared with me and it didn't work.
After that I have tried the 16.43.1 (2020-02-03) version (octokit-rest.min.js) and it's working.
Is there any difference between those files generation (v16.43.1 vs v17.9.2) ??

Thank you very much in advance.

**Reply by gr2m** [reply]
Yes, there are several breaking changes:
https://github.com/octokit/rest.js/releases/v17.0.0

What error do you get? Can you share the code of how you use it?

**Reply by npiccolo** [reply]
In this project I need to import the JS file as a static resource.
When I try to load the 12.9.2 file I just get an 'undefined error'.
But when I load the 16.43.1 file It works fine, I can load it normally.

This is an example of how I load it.
`loadScript(this, octokit + '/index.bundled.js').then(() => {
      console.log('LOADED');
    });`

Using this documentation: https://developer.salesforce.com/docs/component-library/documentation/en/lwc/lwc.js_third_party_library

**Reply by gr2m** [reply]
instead of load script, can you try `import { Octokit } from '...'` instead

**Reply by npiccolo** [reply]
I'm not able to import like that, that's my main restriction (restricted by the platform). 
I would love to do that, would be lot easier !! =D

**Reply by gr2m** [reply]
I see. May I ask what you are trying to build? Importing the whole of `@octokit/rest`, without an deduplication, is currently at 62.7kb minified: https://bundlephobia.com/result?p=@octokit/rest@17.9.3. In comparison [`@octokit/core`](https://github.com/octokit/core.js), which has most of the same functionality but a little less convenience, is at 13kb: https://bundlephobia.com/result?p=@octokit/core@2.5.3

I'll try to figure out the bundling nevertheless :)

**Reply by npiccolo** [reply]
I'm trying to integrate Git operations into Salesforce's LWC Components.
Although the platform is very flexible, it has some restrictions like third-party JS libraries integration.
Basically I need to be able do some Git operations like compare branches, create new branches, create commits and things like that, you think I'll be able to do that kind of operations with @octokit/core or it just proposes rest api calls ?? 

Thank you again for helping me !!! =D

**Reply by gr2m** [reply]
Do you use `octokit.paginate`? Do you expect to run into problems with the rate/abuse limiting?

**Reply by npiccolo** [reply]
I don't think I'll use paginate.
And I don't expect running into problems with the rate/abuse limiting either (I hope !!!) =P

**Comment by gr2m** [comment, marked as answer]
I've created a new bundle and attached it to https://github.com/octokit/rest.js/releases/tag/v17.10.0. Does that one work for you?

**Reply by gr2m** [reply]
for reference, I used `rollup` with the following configuration

```js
const resolve = require("@rollup/plugin-node-resolve").default;
const commonjs = require("@rollup/plugin-commonjs");
const { terser } = require("rollup-plugin-terser");

module.exports = {
  input: "pkg/dist-web/index.js",
  output: {
    file: `pkg/dist-web/octokit.iife.js`,
    format: "iife",
    name: "Octokit",
  },
  plugins: [
    resolve({
      browser: true,
    }),
    commonjs(),
    terser(),
  ],
};
```

**Reply by npiccolo** [reply]
Thanks Gregor !! =D
I'll try it right now !!

**Reply by npiccolo** [reply]
Sorry for my late response !!!
It's working !!!!
Thank you sosososo much !!!
